doc/unicode.md

```diff
@@ -37,35 +54,89 @@ NOT_CYRILLIC : [\P{Script=Cyrillic}] ;
 See [lexer-rules.md](lexer-rules.md#lexer-rule-elements) for more detail on Unicode
 escapes in lexer rules.
 
-# CharStreams and UTF-8
+## Migration
 
-## Java Target
-If your lexer grammar contains code points larger than `U+FFFF`, your
-lexer client code must open the file using `CharStreams.fromPath()` or
-equivalent in your runtime's language, or input values larger than
-`U+FFFF` will *not* match.
 
-For backwards compatibility, the existing `ANTLRInputStream` and
-`ANTLRFileStream` APIs only support Unicode code points up to `U+FFFF`.
+Code for **4.6** looked like this:
 
-The existing `TestRig` command-line interface supports all Unicode
-code points.
 
-## Other Targets
-Other language targets usually have their `ANTLRInputStream` extended to support the full Unicode range. See the target documentation for supported input encodings (e.g. UTF-8) and other related details.
+```java
+CharStream input = new ANTLRFileStream("myinputfile");
```

Comment:
Mention this didn't actually use UTF-8 by default, but actually depended on the calling environment's default (`$LANG` etc.)